check.go

```diff
@@ -727,7 +734,18 @@ func (runner *suiteRunner) forkCall(method *methodType, kind funcKind, testName
 	go (func() {
 		runner.reportCallStarted(c)
 		defer runner.callDone(c)
+		startTime := time.Now()
+
 		dispatcher(c)
+
+		// Don't allow a single to run more than 3 seconds!
+		if c.status() == succeededSt {
+			if runner.methodTimeout > 0 {
+				if time.Since(startTime) > runner.methodTimeout {
```

Comment:
I think we use one `if` is enough. 
```
	if c.status() == succeededSt || runner.methodTimeout > 0 || 
		time.Since(startTime) > runner.methodTimeout {
			....
	}
```

Reply:
PTAL @zimulala